### PR TITLE
Consistent and thinner weight headings

### PR DESCRIPTION
**Pull request by Lukewh**
Fixes https://github.com/canonical-websites/snapcraft.io/issues/1155
Supercedes https://github.com/canonical-websites/snapcraft.io/pull/1147

# QA

- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004
- Visit all the pages affected by this PR (see the file list)
- Ensure the h1's and h2's are the thinner weight

### CODE REVIEW

**Comment by bartaz** [comment]
Categories titles on Store page still use bolder style:

screen shot 2018-10-01 at 09 25 47

**Comment by Lukewh** [comment]
@bartaz those are h3's the thinner weight only applies to h1's and h2's

**Comment by anthonydillon** [comment]
Shouldn't we be managing the title font weights through Vanilla?

**Comment by lyubomir-popov** [comment]
@Lukewh @anthonydillon The thin wait is already in use for h1 headings on ubuntu.com. On snapcraft we never go above h2, so effectively our h2 is functioning as an h1. This is why I made it thin. This has been discussed and agreed with @spencerbygraves and in the branch review meeting last week.

**Comment by spencerbygraves** [comment]
@Lukewh @anthonydillon thin is listed as an option in Vanilla already - https://docs.vanillaframework.io/en/base/typography#font-weights

File: templates/index.html [hunk 1]
```diff
@@ -92,7 +92,7 @@ <h4 class="p-testimonial__title">“A major software discovery tool”</h4>
 <div class="p-strip is-deep is-bordered">
   <div class="row">
     <div class="col-6">
-      <h3>One build for all Linux and IoT</h3>
+      <h3 class="p-heading--two">One build for all Linux and IoT</h3>
```
Comment (bartaz) [inline templates/index.html]: @Lukewh As far as I can see `h3`s get `p-heading-two` class here.

Categories headings on Store page also use `h3`. Shouldn't they get the class as well?

Comment (lyubomir-popov) [reply templates/index.html]: @bartaz No, the categories on /store should get .p-heading--four.

**Comment by lyubomir-popov** [comment]
screen shot 2018-10-02 at 12 09 31
screen shot 2018-10-02 at 12 09 21

Looks good @Lukewh, just one tiny niggle - the spacing under snaps is slightly laerger on /store than /iot. Can you make the iot as big as the store one?

**Comment by Lukewh** [comment]
@lyubomir-popov will do it in a separate PR. Have created https://github.com/canonical-websites/snapcraft.io/issues/1162 to track it.

**Comment by bartaz** [comment]
Snap name on publisher pages got `heading--four` style, which makes main heading the same size as other heading on the page. It looks bit weird:

screen shot 2018-10-03 at 13 59 49

**Comment by bartaz** [comment]
Headings on /build page doesn't seem to be updated

screen shot 2018-10-03 at 14 02 28

**Pull request merged**